crates/storage/store_db/redb.rs

```diff
@@ -767,22 +771,45 @@ impl StoreEngine for RedBStore {
             .map_err(StoreError::RLPDecode)
     }
 
-    fn set_storage_heal_paths(
-        &self,
-        accounts: Vec<(H256, Vec<Nibbles>)>,
-    ) -> Result<(), StoreError> {
-        self.write(
-            SNAP_STATE_TABLE,
-            SnapStateIndex::StorageHealPaths,
-            accounts.encode_to_vec(),
-        )
+    fn set_storage_heal_paths(&self, paths: Vec<(H256, Vec<Nibbles>)>) -> Result<(), StoreError> {
+        let key_values = paths
+            .into_iter()
+            .map(|(hash, paths)| {
+                (
+                    <H256 as Into<AccountHashRLP>>::into(hash),
+                    <Vec<Nibbles> as Into<TriePathsRLP>>::into(paths),
+                )
+            })
+            .collect();
+        self.write_batch(STORAGE_HEAL_PATHS_TABLE, key_values)
```

Comment:
After merging #2336 this needs the following change:
```suggestion
    async fn set_storage_heal_paths(&self, paths: Vec<(H256, Vec<Nibbles>)>) -> Result<(), StoreError> {
        let key_values = paths
            .into_iter()
            .map(|(hash, paths)| {
                (
                    <H256 as Into<AccountHashRLP>>::into(hash),
                    <Vec<Nibbles> as Into<TriePathsRLP>>::into(paths),
                )
            })
            .collect();
        self.write_batch(STORAGE_HEAL_PATHS_TABLE, key_values).await
```
Similar changes will be needed at the API level and for libmdbx.

Reply:
Thanks!

Reply:
Updated with merge!